wnfs/src/private/hamt/diff/node.rs

```diff
@@ -69,33 +69,30 @@ pub struct NodeChange {
 ///    println!("Changes {:#?}", changes);
 /// }
 /// ```
-#[async_recursion(?Send)]
-pub async fn node_diff<K, V, H, B>(
+pub async fn node_diff<K, V, H>(
```

Comment:
Is the node diff now actually used? IIUC, we don't use it for `kv_diff` and we're not using it for `merge` either, right?
I'm thinking we probably don't want to expose it, unless we'd know a specific use case for it? If we delete it now, it's not like it's gone, we can go back and grab it from the history it if it turns out to be useful, but removing it now will make the public API surface smaller & potentially reduce maintenance burden.